src/sign-typed-data.ts

```diff
@@ -206,6 +206,17 @@ function encodeField(
       );
     }
     const parsedType = type.slice(0, type.lastIndexOf('['));
+
+    // If it's a struct, we concatenate their encodedData and then take the keccak256
+    // as per "The array values are encoded as the keccak256 hash of the concatenated encodeData of their contents"
+    if (types[parsedType] !== undefined) {
+      const typeValuePairs = value.map((item) =>
+        encodeData(parsedType, item, types, version),
+      );
+      return ['bytes32', keccak(Buffer.concat(typeValuePairs))];
```

Comment:
Some experimentation has shown me that the solidity behavior of `abi.encode()` when passed an empty array that is typed to hold structs is to return 32 bytes of zero memory.

I have not added a test for this yet, but this is one more way that the current behavior of signing arrays of structs differs from the ecosystem, so it seemed on theme.

```suggestion
    if (typeValuePairs.length === 0) {
        const zero = new Buffer.alloc(32);
        return ['bytes32', zero];
      }

      return ['bytes32', keccak(Buffer.concat(typeValuePairs))];
```